src/models/users

```diff
@@ -0,0 +1,217 @@
+            , 'Giresun', 'Aksaray'
+            , 'Gumushane', 'Bayburt'
+            , 'Hakkari', 'Karaman'
+            , 'Hatay', 'Kirikkale'
+            , 'Isparta', 'Batman'
+            , 'Mersin', 'Sirnak'
+            , 'Istanbul', 'Bartin'
+            , 'Izmir', 'Ardahan'
+            , 'Kars', 'Igdir'
+            , 'Kastamonu', 'Yalova'
+            , 'Kayseri', 'Karabuk'
+            , 'Kirklareli', 'Kilis'
+            , 'Kirsehir', 'Osmaniye'
+            , 'Duzce'
+        ],
+        interests: [{
+            type: string,
+            enum: ['No Poverty', 'Zero Hunger', 'Good Health And Well-Being',
+                'Quality Education', 'Gender Equality', 'Clean Water And Sanitation',
+                'Affordable And Clean Energy', 'Animals', 'Oceans', 'Nature',
+                'Reduced Inequalities', 'Sustainable Cities And Communities', 'Responsible Consumption And Production',
+                'Climate Action', 'Life Below Water', 'Life On Land', 'Peace', 'Youth', 'Justice']
+        }],
+        gender: {
+            type: string,
+            enum: ['male', 'female', 'other']
+        },
+        followedEvents: [{
+            type: mongoose.Schema.Types.ObjectId,
+            ref: "Event",
+
+        }],
+        followedFunds: [{
+            type: mongoose.Schema.Types.ObjectId,
+            ref: "Fund",
+        }],
+        followedUsers: [{
+            type: mongoose.Schema.Types.ObjectId,
+            ref: "User",
+        }],
+        followedOrganizations: [{
+            type: mongoose.Schema.Types.ObjectId,
+            ref: "Organization",
+
+        }]
+
+    }]
+
+})
+
+// virtual
+user.virtual('fullName').get(function () {
+    return this.firstName + ' ' + this.lastName;
+});
+
+const User = BaseUser.discriminator("User", user);
+
+
+
+
+const organization = mongoose.Schema({
+    name: {
+        type: string,
+        required: true
+    },
+    description: {
+        type: string,
+        required: true
+    },
+    coverImage: {
+        type: string,
+        required: true
+
+    },
+    categories: [{
+        type: string,
+        enum: ['No Poverty', 'Zero Hunger', 'Good Health And Well-Being',
+            'Quality Education', 'Gender Equality', 'Clean Water And Sanitation',
+            'Affordable And Clean Energy', 'Animals', 'Oceans', 'Nature',
+            'Reduced Inequalities', 'Sustainable Cities And Communities', 'Responsible Consumption And Production',
+            'Climate Action', 'Life Below Water', 'Life On Land', 'Peace', 'Youth', 'Justice']
+    }],
+    city: {
+        type: string,
+        required: true
+
+    },
+    rates: [rate],
+    // rate : virtual property
+    websiteUrl: {
+        type: string,
+        required: true
+    }
+
+
+});
+
+// virtual
+organization.virtual('rate').get(function () {
+    if (!this.rates.length) return 5
+    return this.rates.flatMap(e => e.rate).reduce((acc, curr) => acc + curr, 0) / rate.length
```

Comment:
I am not sure what are you doing here, but all you have to do is the following: 
`return this.rates.reduce((acc, curr) => acc + curr.rate, 0) / rates.length`

Reply:
i missthought it would be object. fixed tis